dvc/remote/http.py

```diff
@@ -101,7 +101,16 @@ def _request(self, method, url, **kwargs):
         kwargs.setdefault("timeout", self.REQUEST_TIMEOUT)
 
         try:
-            return self._session.request(method, url, **kwargs)
+            res = self._session.request(method, url, **kwargs)
+
+            if kwargs["allow_redirects"] and res.status_code in (301, 302):
+                # AWS s3 doesn't like to add a location header to its redirects
+                # from https://s3.amazonaws.com/<bucket name>/* type URLs.
+                # This should be treated as an error
+                raise requests.exceptions.RequestException
```

Comment:
What if this is not s3 and location header is present?

Reply:
Then `requests` would've followed the redirect.

Reply:
Maybe better to be explicit and check that a location header is not present?

Reply:
Yes, that would be better. At least the comment will follow the code, now it looks like a bug in implementation. So adjusting comment would be good too.

Reply:
Makes sense